sql/core/src/main/scala/org/apache/spark/sql/SQLConf.scala

```diff
@@ -27,6 +27,7 @@ private[spark] object SQLConf {
   val COLUMN_BATCH_SIZE = "spark.sql.inMemoryColumnarStorage.batchSize"
   val IN_MEMORY_PARTITION_PRUNING = "spark.sql.inMemoryColumnarStorage.partitionPruning"
   val AUTO_BROADCASTJOIN_THRESHOLD = "spark.sql.autoBroadcastJoinThreshold"
+  val AUTO_SORTMERGEJOIN = "spark.sql.autoSortMergeJoin"
```

Comment:
Lets make this `spark.sql.planner.sortMergeJoin` and put it below with the other one for external sort.  I say this as there is nothing "automatic" about it based on statistics (like with our broadcast support) and if testing shows this has good performance we might just get rid of the option completely and have it always on.